client/asset/eth/multirpc.go

```diff
@@ -569,12 +569,80 @@ func (m *multiRPCClient) voidUnusedNonce() {
 	m.lastNonce.stamp = time.Time{}
 }
 
-func (m *multiRPCClient) reconfigure(ctx context.Context, settings map[string]string) error {
+// createAndCheckProviders creates and connects to providers. It checks that
+// unknown providers have a sufficient api to trade and saves good providers to
+// file. One bad provider or connect problem will cause this to error.
+func createAndCheckProviders(walletDir string, endpoints []string, net dex.Network,
+	log dex.Logger, ctx context.Context) error {
```

Comment:
It bugs me that the `context.Context` is not the first argument.

Reply:
bugs my linters too

Reply:
Ah, sorry, will move.

Reply:
Hope I haven't done this anywhere else.